src/audio/module_adapter/module/waves/waves.c

```diff
@@ -679,6 +756,9 @@ static int waves_codec_prepare(struct processing_module *mod,
 	if (!ret)
 		ret = waves_effect_setup_config(mod);
 
+	if (!ret)
+		ret = waves_effect_apply_config_blob_cache(mod);
+
```

Comment:
this is not an effective way for such logic, take example
if line 748 return 1, then with current code logic, it have to continue check 5 times ret before real return.
suggest use goto label for each return.

also, line 762 -765 also have logic error, 765 always print.
need else before line 765.